sim/simulator.py

```diff
@@ -156,6 +231,9 @@ def step(self) -> SimulationStatus:
                     print("sim: step error")
                     self._status = SimulationStatus.ERROR
                 elif self._robot_pos[:-1] in self._end:
+                    if self._status != SimulationStatus.IN_PROGRESS_FOUND_DEST:
+                        self._counter.reset_current()
```

Comment:
Not sure resetting the current counters is correct here. The run didn't end yet.